### ISSUE TITLE
Maven Bill Of Meterials (BOM) without transtive dependencies

### ISSUE THREAD

**Issue by RalphEng** [square/okhttp#4407]
A BOM (#4329) should help to keep the version different artifacts of one libary consistent.(https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html) But (IMHO) **it should not contain the used (transitive) artifacts**. 

The BOM `com.squareup.okhttp3:okhttp-bom:3.12.0` does not only provide dependency-management dependencies for `com.squareup.okhttp3` but also a lot more:
- com.squareup.okio:okio
- com.google.code.findbugs:jsr305
- junit:junit
- org.bouncycastle:bcprov-jdk15on
- org.apache.httpcomponents:httpclient
- io.airlift:airline
- com.google.guava:guava
- com.google.android:android
- com.squareup.moshi:moshi
- com.github.jnr:jnr-unixsocket
- org.conscrypt:conscrypt-openjdk-uber

This happend (maybe by accident) because this dependendies are defind in the dependency-management of your parent porm. And this dependency-management is passed (by inheritance) to the bom module.

**Please provide a BOM that contains only `com.squareup.okhttp3` artifacts.**

----------------------------------------
My solution would be having two different modules:

- `com.squareup.okhttp3:okhttp-bom`
- `com.squareup.okhttp3:libdef ` (type pom)  //libary definition

- `com.squareup.okhttp3:libdef` is a new (pom) module that contains a dependency-management (like a bom) intended for internal use. The complete dependency management section is moved from the `:parent` pom to `:libdef ` (libdef maybe also contains a dependency to `:okhttp-bom` in its dependeny-management: 
```
<dependencyManagement>
     <dependencies>
....
          <dependency>
              <groupId>${project.groupId}</groupId>
              <artifactId>okhttp-bom</artifactId>
              <type>pom</type>
              <scope>import</scope>
          </dependency>
```
- `com.squareup.okhttp3:parent` does not contain a dependeny-management at all
- `com.squareup.okhttp3:okhttp-bom` does *not* contains `com.squareup.okhttp3:libdef`
- all "normal" modules import `:libdef` for in its dependency management: 
```
    <dependencyManagement>
       <dependencies>
           <dependency>
              <groupId>${project.groupId}</groupId>
             <artifactId>libdef</artifactId>
             <version>${project.version}</version>
         </dependency>
   </dependencyManagement>
```

**Comment by swankjesse** [comment]
Let's just abandon the bom. Seems like a lot of trouble for marginal benefit.

**Comment by RalphEng** [comment]
IMHO BOMs have a great benefit, should I try provide a Pull request?

**Comment by yschimke** [comment]
@RalphEng Can you point to existing external projects and how they achieve the same?

**Comment by RalphEng** [comment]
For example BOMs provided by Spring (org.springframework:spring-framework-bom5.1.2.RELEASE) provide only dependeies for Spring but not for there transitive dependencies. - But Spring uses Gradle for its build, so they used an other solution for this problem.

An other example is Hibernate OGM, they use slightly different solution: There Parent pom (org.hibernate.ogm:hibernate-ogm-parent:5.4.0.Final) use there bom (org.hibernate.ogm:hibernate-ogm-bom:5.4.0.Final) as its own parent (So they do not need to add an import of the bom in there normal modules).

The solution I described above is used in many of our company's own libaries. So far we have found no disadvantage or problem. Everything works, even maven release,
